qualtran/_infra/bloq.py

```diff
@@ -231,16 +233,36 @@ def add_my_tensors(
         )
         tn.add(qtn.Tensor(data=data, inds=inds, tags=[self.short_name(), tag]))
 
-    def bloq_counts(self, ssa: Optional['SympySymbolAllocator'] = None) -> Set['BloqCountT']:
-        """Return a set of `(n, bloq)` tuples where bloq is used `n` times in the decomposition.
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+        """Override this method to build the bloq call graph.
+
+        This method must return a set of `(n, bloq)` tuples where bloq is called `n` times in
+        the decomposition.
 
         By default, this method will use `self.decompose_bloq()` to count up bloqs.
-        However, you can override this if you don't want to provide a complete decomposition,
-        if you know symbolic expressions for the counts, or if you need to "generalize"
-        the subbloqs by overwriting bloq attributes that do not affect its cost with generic
-        sympy symbols (perhaps with the aid of the provided `SympySymbolAllocator`).
+        However, you can provide specific callees directly by overriding this method if
+        1) you don't want to provide a complete decomposition, 2) you know symbolic expressions
+        for the counts, 3) or you need to "generalize" the subbloqs by overwriting bloq
+        attributes that do not affect its cost with generic sympy symbols using
+        the provided `SympySymbolAllocator`.
         """
-        return self.decompose_bloq().bloq_counts(ssa)
+        return self.decompose_bloq().build_call_graph(ssa)
+
+    def call_graph(
+        self,
+        generalizer: Callable[['Bloq'], Optional['Bloq']] = None,
+        keep: Optional[Sequence['Bloq']] = None,
+        max_depth: Optional[int] = None,
+    ) -> Tuple['nx.DiGraph', Dict['Bloq', Union[int, 'sympy.Expr']]]:
```

Comment:
It's a little confusing that `build_call_graph` is supposed to return `(n, bloq)` tuples but the `call_graph` returns `(bloq, n)`. Maybe let's change to `(bloq, n)` at both the places to keep things consistent?

Reply:
yeah good point.

The first is tuples so it doesn't really matter and the latter is a dictionary where it does matter and bloq should be key; so sadly we have to make a change. But I agree that they should be consistent.